### PR TITLE
fixes #78: Extended unicode

### LINKED ISSUE #78: ES6: extended unicode escapes in strings, templates, and regexps "\u{123456}"

### PR DESCRIPTION
**Pull request by emattson**
likely needs a few more tests before merge

### CODE REVIEW

File: test/incompatibilities.js [hunk 1]
```diff
@@ -127,4 +127,13 @@ suite("Parser", function() {
       new Shift.ArrayExpression([new Shift.LiteralNumericExpression(1), new Shift.LiteralNumericExpression(2)]), 
       new Shift.ExpressionStatement(new Shift.LiteralNumericExpression(0))));
   });
+
+  // ES5: allows only `\u Hex4Digits`
+  // ES6: allows `\u Hex4Digits` and `\u {HexDigits}`
```
Comment (michaelficarra) [inline test/incompatibilities.js]: These tests shouldn't go into incompatibilities, since `\u{xxxx}` style escapes were syntax errors in ES5.

**Comment by michaelficarra** [comment]
I've rebased off the es6 branch for you (and squashed while I was at it). Please reset --hard to origin/extended-unicode before continuing work on this.

File: src/tokenizer.js [hunk 2]
```diff
@@ -769,6 +745,51 @@ export default class Tokenizer {
     return r1 << 4 | r2;
   }
 
+  scanUnicode() {
```
Comment (ikarienator) [inline src/tokenizer.js]: This method looks a bit too long for its purpose. Can you abstract it a little bit (say turn the second one into a loop, then the two branches can be merged)?

Comment (emattson) [inline src/tokenizer.js]: Method could be shortened. However, I'm recommending  we leave it as is in order to maintain readability of the code.

**Pull request merged**